ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -173,6 +201,143 @@ S3GetDirectory(const std::string &home, const std::string &name, bool create, st
     return (ret);
 }
 
+/*
+ * S3FileClose --
+ *    File handle close.
+ */
+static int
+S3FileClose(WT_FILE_HANDLE *fileHandle, WT_SESSION *session)
+{
+    int ret;
+    S3_FILE_HANDLE *s3FileHandle = (S3_FILE_HANDLE *)fileHandle;
+    S3_STORAGE *storage = s3FileHandle->storage;
+
+    /* Remove the file handle from the list. */
+    {
```

Comment:
why {}?

Reply:
I guess the `lock_guard` is destroyed when it goes beyond the scope and the mutex is then released. Correct me if I am wrong @bingx1.
Would it hurt to put `lock_guard` around the whole function?

Reply:
Correct - I was scoping the `lock_guard` - though given there isn't much left to do in the function following this point it may be okay to omit. I wasn't sure what was best price. We could put `lock_guard` around the whole function - I don't see why not; though does it add any value?

Reply:
I'm not a fan to implicitly release locks. Can we explicitly do that? If not, may be we should add a comment there.

Reply:
I would define the `lock_guard` at the very beginning of the function and it will get deleted when the function ends. This is something we do in our cpp test suite and it has been useful.
I understand what you mean by `implicit` @quchenhao, but this is what the `lock_guard` was designed for, this is probably the most common use case.

Reply:
I'm not sure why we would need to increase the length of the critical section any longer than necessary. Is there any reason why? If I remove the curly braces that are currently scoping the lock, the `lock_guard` will be scoped by the function anyway and will be unlocked once the program leaves the function scope

Reply:
Agreed, relying on manual unlocking is not recommended - lock_guards are useful and the recommended way to safeguard against deadlocks. Also, agreed with Bing, to limit the critical section to the minimum needed. It also safeguards against adding more functionality to the method tomorrow, increasing the critical section unintendedly.

Reply:
I agree with Chenhao, a comment would help explain what is going on here.

Reply:
I've added a comment to make it clear that the lock is being unlocked once the scope is exited.

Reply:
Where is the comment? Have you pushed it?

Reply:
My bad - I only added one to the other place (line 336) where I was using a `lock_guard`. I will add one here too.

Reply:
I think we should. Just a short one though.